### ISSUE TITLE
Drop python 3.7 support

### ISSUE THREAD

**Issue by tinosulzer** [pybamm-team/PyBaMM#2370]
It has now been dropped by numpy https://numpy.org/devdocs/release/1.22.0-notes.html

**Comment by abillscmu** [comment]
since the Julia stuff is failing tests on 3.7 anyway does it make sense to just put it there?

**Comment by tinosulzer** [comment]
Would be cleaner as a separate PR

**Comment by agriyakhetarpal** [comment]
Hi @tinosulzer, could I work on this one?

**Comment by tinosulzer** [comment]
Yes, but we need it for #2363 , so needs to be done quickly (next couple of days)

**Comment by tinosulzer** [comment]
You just need to find where "3.7" appears in the repo, and
- if it's in [3,7, 3.8, 3.9], remove it
- if it's on its own, change it to 3.9
